nodejs/nodejs.org
# fix: language selector from not closing

## Description

This PR updates the `Dropdown` component to accept `children` as the wrapping elements of the Dropdown. (It is important to mention that right now the Dropdown component has no self-aligning logic/styles and that should probably be done)

It also updated the `useClickOutside` Hook that was registering a new click handler on every re-render, which is definitely not a good thing.

Finally, it updates the `LanguageSelector` component with the new usage of `Dropdown` and updates the `handle/callback` of `useClickOutside` to only update the state to `false` if the state is `true` (avoid unnecessary state updates) and it prevents propagation of the onClick event to other click handlers if it comes from the `buttons` `onClick` handler.

## Validation

This was tested through a Storybook. This should be validated by other reviewers before approving the PR.

- Open the Storybook
- Go to "New Header"
- Click on Language selector icon
  - It should open the Dropdown
- Click outside of Language Selector
  - It should close the Dropdown
- Click on Language selector icon (again)
  - It should open
- Click on the Language selector icon (again 2)
  - It should close

## Related Issues

- Closes https://github.com/nodejs/nodejs.org/issues/5509
- Closes https://github.com/nodejs/nodejs.org/issues/5508

### Check List



- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
